### PR TITLE
[website] update Styleguide package, tweak UI themes

### PR DESCRIPTION
**Pull request by Simek**
# Why

Currently, Snack website is using a very old version of our Styleguide package, let's update it.

# How

Switch to the `styleguide-base` variant, since more robust `styleguide` requires now Next app and it's based on context and component from Next. Definitely there is a lot room for the improvements, that's why it's a draft for now, but at least we can run Snack locally after the changes.

Since we now using Radix UI palettes as a based for ours and Monaco doesn't support `hsl` color definition which Radix uses I have to introduce the conversion helper, which allow us to stick to the palette, while making colors compatible with Monaco.

# Test Plan

The changes have been tested by running Snack stack from root directory (`yarn start`).

# Preview

Screenshot 2023-05-17 at 16 10 21
Screenshot 2023-05-28 at 18 42 08

### CODE REVIEW

File: website/src/client/components/ThemeProvider.tsx [hunk 1]
```diff
@@ -142,3 +135,16 @@ const styles = StyleSheet.create({
...
 });
+
+function convertToHex(color: string) {
+  if (color.startsWith('hsl')) {
+    const hslChunks = color
+      .replace('hsl(', '')
+      .replace(')', '')
+      .replace('%', '')
+      .split(',')
+      .map((c) => parseInt(c.trim(), 10));
```
Comment (byCedric) [inline website/src/client/components/ThemeProvider.tsx:146]: I think using a RegExp might be better here, just to support slightly different notations (such as `hsl (...)` or rogue spaces).

From what I could find online, [this regex seems to do exactly that](https://regex101.com/r/ZSUEig/1) - based on this lib: https://github.com/regexhq/hsl-regex/blob/master/index.js#LL8C5-L8C67 (with some slight adjustments)

**Review by byCedric — COMMENTED** [review]
Looking good so far 😄

File: website/package.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
   },
   "dependencies": {
     "@babel/parser": "^7.12.16",
-    "@expo/styleguide": "^1.1.0",
+    "@expo/styleguide-base": "^1.0.0-alpha.3",
```
[the remark this answers is not in the document]
Comment (Simek) [reply website/package.json:31]: There are no stable version yet, might release one soon tho.

**Review by byCedric — APPROVED** [review]
LGTM!

**Pull request merged**